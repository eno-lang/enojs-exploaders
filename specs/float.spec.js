const eno = require('enojs');
const loaders = require('../loaders.js');

describe('float', () => {

  let document;

  describe('42', () => {
    it('returns 42.0', () => {
      document = eno.parse('float: 42');
      expect(document.field('float', loaders.float)).toBe(42.0);
    });
  });

  describe('-42', () => {
    it('returns -42.0', () => {
      document = eno.parse('float: -42');
      expect(document.field('float', loaders.float)).toBe(-42.0);
    });
  });

  describe('42.0', () => {
    it('returns 42.0', () => {
      document = eno.parse('float: 42.0');
      expect(document.field('float', loaders.float)).toBe(42.0);
    });
  });

  describe('42,0', () => {
    it('throws an error', () => {
      document = eno.parse('float: 42,0');
      expect(() => document.field('float', loaders.float)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('4 2.0', () => {
    it('throws an error', () => {
      document = eno.parse('float: 4 2.0');
      expect(() => document.field('float', loaders.float)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('fortytwo dot zero', () => {
    it('throws an error', () => {
      document = eno.parse('float: fortytwo dot zero');
      expect(() => document.field('float', loaders.float)).toThrowErrorMatchingSnapshot();
    });
  });
});
