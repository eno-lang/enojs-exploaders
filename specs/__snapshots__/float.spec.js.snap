// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`float 4 2.0 throws an error 1`] = `
"'float' must be a float.

   Line | Content
 >    1 | float: 4 2.0
"
`;

exports[`float 42,0 throws an error 1`] = `
"'float' must be a float.

   Line | Content
 >    1 | float: 42,0
"
`;

exports[`float fortytwo dot zero throws an error 1`] = `
"'float' must be a float.

   Line | Content
 >    1 | float: fortytwo dot zero
"
`;
